### DISCUSSION TITLE
EDI menu is missing in build and run.

### DISCUSSION THREAD

**Discussion by infinedi** [blueseerERP/blueseer#122] [General]
When I build and run your code in netbeans, the EDI menu is missing.  How do I include it in the build and debug?  I'd like to study how it works.

**Comment by vaughnte** [comment]
yes...that's an old refresh.bat script with an old bsdbdev.db database file.   I'll get those updated.   In the meantime, download the file from the following link....and overwrite the test/data/bsdb.db file in your netbeans project 'working directory' (see project properties under 'run') with the bsdbdev.db file.   
Download:  https://github.com/blueseerERP/blueseer/blob/master/sf/data/bsdbdev.db .   
Make sure you are left with test/data/bsdb.db filename in the working directory.

**Comment by infinedi** [comment]
Awesome, that fixed it.  ' Thanks !!!

Arthur Didion
IT Representative
***@***.***

**Comment by infinedi** [comment]
The EDI menu is working, but, I ran into another issue.  On the EDI editor, when I load a map, like bs850ToIDOC, the IFS and OSF load fine, but the Map file doesn't.  The editor quit working.

***@***.***

Thank You...


Arthur Didion
IT Representative
***@***.***

**Reply by vaughnte** [reply]
Other than modifying the layout of each menu....I've never really used NetBeans for actual execution...I've always used ant.   The 'test' working directory doesn't have a 'lot' of the internal directories necessary to drive it to full execution.   You can remedy that by downloading the actual application .exe...foun here:
https://github.com/blueseerERP/blueseer/releases/download/v6.6/blueseer.sqlite.win.v6.6.exe

...and installing it into c:\blueseertest folder.   Then set your 'working directory' for the netbeans project to that directory...c:\blueseertest.   It will have all the 'extras' necessary to fully exercise the application runtime.
That's a workaround.   WIth that said, I should probably update the 'refresh.bat' script to establish a legitimate 'test' working directory for those using netbeans to both adjust menu layouts and actual execution of the application.    I'll put it on the list.

**Comment by infinedi** [comment]
Thanks for the solution on how to get up and running so I could test some code.  I'm studying your EDI system and I'm a bit curious how it works.  We do EDI processing here at my current job -- insurance claim processing, ANSI837 is our big one.  I'm hoping to learn from your code and maybe use some of your techniques.  I was experimenting with your software, testing some of the features, and noticed that the sample files are not valid ANSI X12.  I tried a valid X12 850 that I found off the internet, and ran your code against it and it failed:  " Establish isaMAP...check structure of input...possible double delimiters" error.  I got to looking at your sample file and it's not valid ANSI X12.  Your sample file doesn't have all the parsing delimiters in the ISA ( * > ~ ) and all the segments don't terminate with a ~.   Going through your code, it looks like the segment delimiter is keying off an empty space ' ' or CR/LF if they exist (that I think if both CR and LF do exist, that it might write two of the same lines... not for sure, just doing rough code scan.).  Anyway, I'm not sure that the sample is fully flushed out, which is OK if your just showing how the process works.  The mapping process looks pretty sound.  I'm going to do some more testing to see if I can make my own IFS and OFS to see if I can get it to work.  I'm really excited about your code and really appreciate an opportunity to learn from it.  Also, thanks for your quick replies.  I hope I can share something of my EDI experience with you too to give back in kind, if I can.  Thoughts?  Thanks, Art...


Here is a valid 850 X12 file I got off the internet:

ISA*00*          *00*          *16*SENDER1        *1B*RECEIVER1      *071216*1406*U*00204*000000263*1*T*>~
GS*IN*SENDER1*RECEIVER1*20071216*1406*000000001*X*004010~
ST*850*0001~
BEG*00*SA*XX-1234**20170301**NA~
PER*BD*ED SMITH*TE*8001234567~
TAX*53247765*SP*CA*********9~
N1*BY*ABC AEROSPACE*9*1234567890101~
N2*AIRCRAFT DIVISION~
N3*2000 JET BLVD~
N4*FIGHTER TOWN*CA*98898~
PO1*1*25*EA*36*PE*MG*XYZ-1234~
MEA*WT*WT*10*OZ~
IT8*******B0~
SCH*25*EA***106*20170615~
CTT*1~
AMT*TT*900~
SE*15*0001~
GE*1*000000001~
IEA*1*000000263~


Here is the sample file in the Blueseer sample data folder.

ISA*00*          *00*          *ZZ*999999999      *ZZ*BLUESEER       *210702*1444*U*00400*001150061*0*P*~
GS*PO*999999999*BLUESEER*20210702*1444*1150061*X*004010
ST*850*0001
BEG*00*NE*PONBR123**20210702
FOB*DF*ZZ*VENDOR
DTM*002*20210702
N1*ST*BLUESEER DEPT*92*111
N3*WILEY LOOP ROAD
N4*KINGSPORT*TN*37662
N1*BT*BLUESEER DEPT*92*222
N3*945 RAEFORD BLVD
N4*DILLON*SC*29536
PO1*1*260*PC*66.48**VP*433332*BP*BR549
PID*F****BICYCLE
PO1*2*208*PC*12.768**VP*33331*BP*BR222
PID*F****TRICYCLE
CTT*2*468
SE*16*0001
GE*1*1150061
IEA*1*001150061

Arthur Didion
IT Representative
***@***.***

**Reply by vaughnte** [reply]
The sample file (ACME_850.txt) is indeed valid X12.   X12 supports a variety of segment, element, and sub-element delimiters and are not limited to a single segment delimiter.   The segment delimiter in the sample file is NL (or hex 0A).   The element delimiter is * and the subelement delimiter (repitition delim) is ~ .     The delimiters exhanged with your trading partner are typically defined by a mutually agreed upon Spec file...usually the larger trading partner.    The sample maps provided work as expected, but if you come across a demo map operating on demo data (as provided in github) that is not working let me know.   With regards to the 837 you are attempting...send me an email at support@blueseer.com ....and we can discuss your requirement further.    I would be interested in developing a demo 837 map if I know the object format that is being translated to/from .
